src/tools/AbstractSelectionTool.tsx

```diff
@@ -0,0 +1,51 @@
+/**
+ * TODO DB: Selection logic for mobile - to discuss with CS
+ * 
+ * @author Daniela Buzatu
+ */
+export class AbstractSelectionTool {
+  /**
+   * If the host components needs access to the selected items it should use this property
+   */
+  selectedItems: string[] = [];
+
+  /**
+     * The host component should listen when the selection changes by passing this function
+     * (for updating the item renderers).
+     */
+  selectionChangedHandler: () => void;
+
+  /**
+   * Should be called by the host component when an item is clicked or right clicked (contextMenu event) or when user clicks outside any selectable items (for reseting the selection),
+   * or any other user action intended to select items happens
+   * (e.g. in gantt: drawing selection rectangle)
+   * It adds/removes from selection (in case of multiple selection) or set the selection (in case of single selection)
+   * 
+   * Usually the right click on the host component works just as a left click regarding the selection.  So the host component calls should be the same in both click cases 
+   * TODO DB: ask CS: in windows explorer right click works different in one special case: ctrl + simple right click (but not shift + simple right click, and not ctrl + right click selection rectangle)
+   */
+  addRemoveItems(itemsKeys: string[], isCtrlOrshiftPressed: boolean) {
```

Comment:
exista in API JS metoda sa luam state-ul curent al keyboard, fara sa le luam din event? Am cautat rapid pe net, si nu am gasit. Toata lumea vorbeste de handleri. 

In cazul tau, de peste tot de unde apelam, ai la dispozitie un event, nu? Poate ar fi mai bine sa dam ca parametru event-ul in cazul asta. Poate pe viitor vom decide logica diferita pt shift si ctrl.

Reply:
Nu exista decat prin event. Am modificat sa accepte event-ul